### PR TITLE
Handling index removal from namespace

### PR DESCRIPTION
**Pull request by tanmayja**
When a mount point from a namespace index has been removed, that namespace has to go to cold restart by server to re-build the index for the same. So in operator when a mount point from index will get removed, operator will delete all the files from all mount points in that particular namespace.

### CODE REVIEW

File: controllers/rack.go [hunk 1]
```diff
@@ -540,6 +540,12 @@ func (r *SingleClusterReconciler) upgradeRack(
 
 		podNames := getPodNames(podsBatch)
 
+		// For each block volume removed from a namespace, pod status dirtyVolumes is appended with that volume name.
```
Comment (sud82) [inline controllers/rack.go:543]: I think we should put this function inside deletePodAndEnsureImageUpdated function, which is called after checking for migrations, quiesce.

File: controllers/rack.go [hunk 2]
```diff
@@ -747,16 +747,17 @@ func (r *SingleClusterReconciler) rollingRestartRack(
...
 		podsBatch := podsBatchList[0]
 
-		res := r.rollingRestartPods(rackState, podsBatch, ignorablePods, restartTypeMap)
-		if !res.isSuccess {
-			return found, res
-		}
-
 		// For each block volume removed from a namespace, pod status dirtyVolumes is appended with that volume name.
 		// For each file removed from a namespace, it is deleted right away.
 		if err := r.handleNSOrDeviceRemoval(rackState, podsBatch); err != nil {
```
Comment (sud82) [inline controllers/rack.go:752]: This we can put inside `restartPods()` function

**Review by sud82 — CHANGES_REQUESTED** [review]

File: api/v1beta1/aerospikecluster_validating_webhook.go [hunk 3]
```diff
@@ -336,10 +337,17 @@ func (c *AerospikeCluster) validateRackUpdate(
 				}
 
 				if len(oldRack.AerospikeConfig.Value) != 0 || len(newRack.AerospikeConfig.Value) != 0 {
+					var statusRack Rack
```
Comment (sud82) [inline api/v1beta1/aerospikecluster_validating_webhook.go:340]: statusRack will become the last rack from c.Status.RackConfig.Racks.

File: controllers/rack.go (hunk 1 above)
Comment (tanmayja) [reply controllers/rack.go]: done

File: api/v1beta1/aerospikecluster_validating_webhook.go (hunk 3 above)
Comment (tanmayja) [reply api/v1beta1/aerospikecluster_validating_webhook.go]: fixed

**Review by abhishekdwivedi3060 — APPROVED** [review]

**Review by sud82 — APPROVED** [review]

**Pull request merged**